src/pipe_wrap.cc

```diff
@@ -241,6 +241,12 @@ void PipeWrap::Connect(const FunctionCallbackInfo<Value>& args) {
                      *name,
                      AfterConnect);
 
+  TRACE_EVENT_NESTABLE_ASYNC_BEGIN1(TRACING_CATEGORY_NODE2(net, native),
```

Comment:
It looks strange to me to spread the begin and end of the same event in different components. Maybe we can add a new base method `Connect` to the `ConnectionWrap`, in which we can track the beginning of the connect event?

Reply:
thanks for suggestions! Is the code like this ?

`connection_wrap.h`
```
template <typename... Args>
static int ConnectTo(ConnectWrap* req_wrap, Args... args);
```
`connection_wrap.cc`
```
template <typename WrapType, typename UVType>
template <typename... Args>
int ConnectionWrap<WrapType, UVType>::ConnectTo(ConnectWrap* req_wrap, Args... args) {
  // trace here
  return req_wrap->Dispatch(args...);
}

using uvPipeConnect = void (*)(uv_connect_t* req, uv_pipe_t* handle, const char* name, uv_connect_cb cb);
using uvTcpConnect = int (*)(uv_connect_t* req, uv_tcp_t* handle, const struct sockaddr* addr, uv_connect_cb cb);

template int ConnectionWrap<PipeWrap, uv_pipe_t>::ConnectTo(ConnectWrap* req_wrap,
                                                            uvPipeConnect,
                                                            uv_pipe_t* handle,
                                                            const char* name,
                                                            uv_connect_cb cb);

template int ConnectionWrap<TCPWrap, uv_tcp_t>::ConnectTo(ConnectWrap* req_wrap,
                                                          uvTcpConnect,
                                                          uv_tcp_t* handle,
                                                          const sockaddr* addr,
                                                          uv_connect_cb cb);
```
`tcp_wrap.cc`
```
ConnectTo(req_wrap,
          uv_tcp_connect,
          &wrap->handle_,
          reinterpret_cast<const sockaddr*>(&addr),
          AfterConnect);
```
`pipe_wrap.cc`
```
ConnectTo(req_wrap,
          uv_pipe_connect,
          &wrap->handle_,
          reinterpret_cast<const char*>(*name),
          AfterConnect);
```

Reply:
Yeah, this might be superfluous just for tracing. As ConnectionWrap just has two subclasses, maybe it's fine to duplicate the traces. Thank you for your patience!